mne/channels/channels.py

```diff
@@ -235,6 +237,87 @@ def get_channel_types(self, picks=None, unique=False, only_data_chs=False):
         return _get_channel_types(self.info, picks=picks, unique=unique,
                                   only_data_chs=only_data_chs)
 
+    def get_ch_positions(self, picks=None):
+        """Get a dictionary of channel positions for each channel.
+
+        Channel positions are in meters in the specified coordinate frame
+        of the montage. All channel positions are a 3-dimensional vector
+        corresponding to xyz positions in meters. For MEG channels, they
+        also add the vectorized orientation.
+
+        For more information on the different channel type
+        positions obtained, see :class:`mne.Info` documentation.
+
+        Parameters
+        ----------
+        picks : str | list | slice | None
+            None corresponds to all data channels that are not MEG.
+
+        Returns
+        -------
+        ch_positions : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding list of xyz coordinates in the montage
+             coordinate frame.
+        ch_frame_dict : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding montage coordinate frame.
+        """
+        # only get channels that were picked
+        picks = _picks_to_idx(self.info, picks)
+        chs = [self.info['chs'][ipick] for ipick in picks]
+
+        # extract name, position, orientation (for MEG)
+        ch_names, ch_pos = [], []
+        for ch in chs:
+            if ch['kind'] == FIFF.FIFFV_MEG_CH:
+                pos = ch['loc']
+            else:
+                pos = ch['loc'][:3]
```

Comment:
I wouldn't triage like this, I would just always take `[:3]`. The remaining terms are not a position but an orientation, and they aren't even in a particularly human-readable form. We should find some other way to get people that info about MEG channels if they want it.

Also, one sneaky problem that we should be careful about is copies vs views. Here `info` is never copied, `info['chs'][ii]` i.e. `ch` is a dict, `ch['loc']` is an ndarray, and `ch['loc'][:3]` is a view because it's a slice. So someone doing on the output `ch_pos_dict['EEG 001'][:] = np.nan` will set `inst.info['chs'][ii]['loc'][:3] = np.nan` effectively. So better here to do:
```suggestion
            pos = ch['loc'][:3].copy()
```

Reply:
Good catch! thanks.